js/app/viewmodel.js

```diff
@@ -45,25 +45,42 @@ class CompletionManager {
     this.rx_pinned.onNext(this.pinned);
   }
 
+  validateArray(newCompletion){
+    if (!Array.isArray(newCompletion)) {
+      window.alert("Error: Invalid fishing checklist.");
+      return false;
+    }else{
```

Comment:
To keep style consistent, would you mind using spaces before "{" and after "}".
```suggestion
    } else {
```